### PR TITLE
32-bit support

### PR DESCRIPTION
**Pull request by Goldfish64**
32-bit support under 10.4 to 10.7

- [x] Enable 32-bit compilation
- [x] 32-bit project scheme
- [x] 32-bit CI builds
- [x] 32-bit kernel macho
- [x] 10.4 and 10.5 kext load hooking and patching
- [x] strlcpy implementation and ml_get_interrupts_enabled implementation for 32-bit
- [x] Kauth root mount hook to replace MAC framework hook
- [x] EFI64 support and tested
- [x] EFI32 support
- [x] EFI32 tested
- [x] Prelinked kernel support under 10.6 (fuzzy and KXLD) - **won't fix**
- [x] Prelinked kernel support under 10.7
- [x] Verification that binary in memory matches on disk binary on 10.4 - no UUIDs in 10.4 - **won't fix**
- [x] RTC read/write - no IOUserClient::externalMethod in 10.4 - **not available when running on 10.4**
- [x] CPU topology - no pmKextRegister in 10.4 - **not available when targeting 10.4/32-bit builds**
- [x] Boot argument parsing and checking - no PE_parse_boot_argn on 10.4
- [x] Fix division of 64-bit values (generates __udivdi3 by default) - **implemented in https://github.com/acidanthera/MacKernelSDK/commit/529a872c2119ef3164fa69457bfb59bca2758de0**
- [x] Long function routing
- [ ] **This kext uses obsolete symbol appears** during boot on 10.6, caused by N_NO_DEAD_STRIP set on certain external symbols in plugins

### CODE REVIEW

File: Lilu/Headers/kern_disasm.hpp [hunk 1]
```diff
@@ -54,7 +61,14 @@ class Disassembler {
 	EXPORT static size_t quickInstructionSize(mach_vm_address_t ptr, size_t min);
 
 	/* Note, code should point to at least 32 valid bytes. */
+#if defined(__i386__)
+	EXPORT static size_t hdeDisasm(mach_vm_address_t code, hde32s *hs);
```
Comment (vit9696) [inline Lilu/Headers/kern_disasm.hpp:65]: In my opinion, go with `using` to choose the local struct.

File: Lilu/Headers/kern_mach.hpp [hunk 2]
```diff
@@ -20,6 +20,22 @@
 #include <mach/vm_param.h>
 #include <libkern/c++/OSDictionary.h>
 
+#if defined(__i386__)
+#define mach_header_current			mach_header
```
Comment (vit9696) [inline Lilu/Headers/kern_mach.hpp:24]: Do not use defines, go with using.

File: Lilu/Info.plist [hunk 3]
```diff
@@ -35,12 +35,27 @@
...
+		<key>com.apple.kpi.libkern</key>
+		<string>8.0.0</string>
+		<key>com.apple.kpi.mach</key>
+		<string>8.0.0</string>
+		<key>com.apple.kpi.unsupported</key>
+		<string>8.0.0</string>
+		<key>com.apple.kernel.6.0</key>
+		<string>7.9.9</string>
+	</dict>
+	<key>OSBundleLibraries_x86_64</key>
```
Comment (vit9696) [inline Lilu/Info.plist:58]: Does OpenCore support this?

File: Lilu/Sources/kern_mach.cpp [hunk 4]
```diff
@@ -234,23 +236,31 @@ mach_vm_address_t MachInfo::findKernelBase() {
...
 			// 0xC is MH_FILESET, available exclusively in newer SDKs.
 			if (getKernelVersion() >= KernelVersion::BigSur && header->filetype == 0xC && header->flags == 0 && header->reserved == 0) {
 				DBGLOG("mach", "found kernel nouveau mach-o header address at %llx", tmp);
 				m_kernel_collection = kernel_collection = true;
 				break;
 			}
-			auto segmentCommand = reinterpret_cast<segment_command_64 *>(header + 1);
+#endif
+			
+			// 10.5 and older have __PAGEZERO first and __TEXT second.
```
Comment (vit9696) [inline Lilu/Sources/kern_mach.cpp:251]: Maybe search by name?

File: Lilu/Sources/kern_patcher.cpp [hunk 5]
```diff
@@ -387,6 +400,14 @@ mach_vm_address_t KernelPatcher::routeFunctionInternal(mach_vm_address_t from, m
 		absolute = true;
 	}
 
+#if defined(__i386__)
+	if (jumpType != JumpType::Auto && jumpType != JumpType::Short) {
```
Comment (vit9696) [inline Lilu/Sources/kern_patcher.cpp:404]: Not sure this is right, as we want long jumps explicitly in many places in Lilu. Perhaps need to force short on i386, but I am really unsure.

File: Lilu/Info.plist (hunk 3 above)
Comment (Goldfish64) [reply Lilu/Info.plist:58]: Implemented in https://github.com/acidanthera/OpenCorePkg/commit/91439cb72d4de608a2f22121f5ab4d12141dc39a.

File: Lilu/Headers/kern_disasm.hpp (hunk 1 above)
Comment (Goldfish64) [reply Lilu/Headers/kern_disasm.hpp]: Corrected in https://github.com/acidanthera/Lilu/commit/5041831557fbde00ab343a608614a180b4ee0f62

**Pull request merged**